File: html/data/update_navigation.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""批量更新所有 quiz 文件的 navigation URL"""

import os
import re
from pathlib import Path

def update_file(filepath):
    """更新单个文件中的 URL"""
    try:
        # 读取文件内容
        with open(filepath, 'r', encoding='utf-8') as f:
            content = f.read()
        
        # 替换模式：将 "XX-name-quiz.html" 替换为 "quiz.html?chapter=XX"
        # 匹配格式：两位数字-任意单词字符和连字符-quiz.html
        pattern = r'"(\d{2})-[\w-]+-quiz\.html"'
        replacement = r'"quiz.html?chapter=\1"'
        
        # 执行替换
        new_content = re.sub(pattern, replacement, content)
        
        # 如果内容有变化，写回文件
        if new_content != content:
            with open(filepath, 'w', encoding='utf-8', newline='') as f:
                f.write(new_content)
            return True
        return False
    except Exception as e:
        print(f"❌ 处理文件 {filepath} 时出错: {e}")
        return False

def main():
    """主函数"""
    # 获取当前目录
    current_dir = Path(__file__).parent
    
    # 查找所有 quiz-*.js 文件
    quiz_files = sorted(current_dir.glob('quiz-*.js'))
    
    if not quiz_files:
        print("❌ 未找到任何 quiz-*.js 文件")
        return
    
    print(f"📁 找到 {len(quiz_files)} 个文件")
    print("=" * 50)
    
    updated_count = 0
    skipped_count = 0
    
    for filepath in quiz_files:
        filename = filepath.name
        if update_file(filepath):
            print(f"✅ 已更新: {filename}")
            updated_count += 1
        else:
            print(f"⏭️  跳过: {filename} (无需更新)")
            skipped_count += 1
    
    print("=" * 50)
    print(f"✨ 完成！更新了 {updated_count} 个文件，跳过 {skipped_count} 个文件")

if __name__ == '__main__':
    main()

File: html/data/update_navigation.js
#!/usr/bin/env node
/**
 * 批量更新所有 quiz 文件的 navigation URL
 */

const fs = require('fs');
const path = require('path');

/**
 * 更新单个文件
 */
function updateFile(filepath) {
    try {
        // 读取文件内容
        const content = fs.readFileSync(filepath, 'utf8');
        
        // 替换模式：将 "XX-name-quiz.html" 替换为 "quiz.html?chapter=XX"
        const pattern = /"(\d{2})-[\w-]+-quiz\.html"/g;
        const newContent = content.replace(pattern, '"quiz.html?chapter=$1"');
        
        // 如果内容有变化，写回文件
        if (newContent !== content) {
            fs.writeFileSync(filepath, newContent, 'utf8');
            return true;
        }
        return false;
    } catch (error) {
        console.error(`❌ 处理文件 ${filepath} 时出错:`, error.message);
        return false;
    }
}

/**
 * 主函数
 */
function main() {
    const currentDir = __dirname;
    
    // 查找所有 quiz-*.js 文件
    const files = fs.readdirSync(currentDir)
        .filter(file => /^quiz-\d{2}\.js$/.test(file))
        .sort();
    
    if (files.length === 0) {
        console.log('❌ 未找到任何 quiz-*.js 文件');
        return;
    }
    
    console.log(`📁 找到 ${files.length} 个文件`);
    console.log('='.repeat(50));
    
    let updatedCount = 0;
    let skippedCount = 0;
    
    files.forEach(file => {
        const filepath = path.join(currentDir, file);
        if (updateFile(filepath)) {
            console.log(`✅ 已更新: ${file}`);
            updatedCount++;
        } else {
            console.log(`⏭️  跳过: ${file} (无需更新)`);
            skippedCount++;
        }
    });
    
    console.log('='.repeat(50));
    console.log(`✨ 完成！更新了 ${updatedCount} 个文件，跳过 ${skippedCount} 个文件`);
}

// 运行主函数
main();
